components/datatypes/consentpreferences.schema.json

```diff
@@ -0,0 +1,424 @@
+        "not_applicable":"Not Applicable"
+      }
+    },
+    "basisOfProcessing": {
+      "title": "Basis Of Processing",
+      "type": "string",
+      "description": "Legal basis for collecting/processing data. Is user consent required?",
+      "enum": [
+        "consent",
+        "legitimate_interest",
+        "contract",
+        "compliance",
+        "vital_interest",
+        "public_interest"
+      ],
+      "meta:enum": {
+        "consent": "User Consent",
+        "legitimate_interest": "Legitimate Interest",
+        "contract": "Contract",
+        "compliance": "Compliance with a Legal Obligation",
+        "vital_interest": "Vital Interest of the Individual",
+        "public_interest": "Public Interest"
+      }
+    },
+    "timestamp": {
+      "title": "Preference timestamp",
+      "description": "Timestamp of this specific opt out or preference.",
+      "type": "string",
+      "format": "date-time"
+    },
+    "source": {
+      "title": "Source of Consent",
+      "description": "Through which feature/product/method did the user provide this consent value",
+      "type": "string",
+      "maxLength": 20
+    },
+    "consentOption": {
+      "type": "object",
+      "properties": {
+        "xdm:choice": {
+          "$ref": "#/definitions/consentValue"
+        },
+        "xdm:basisOfProcessing": {
+          "$ref": "#/definitions/basisOfProcessing"
+        },
+        "xdm:timestamp": {
+          "$ref": "#/definitions/timestamp"
+        },
+        "xdm:source": {
+          "$ref": "#/definitions/source"
+        }
+      }
+    },
+    "marketingOption": {
+      "type": "object",
+      "properties": {
+        "xdm:choice": {
+          "$ref": "#/definitions/consentValue"
+        },
+        "xdm:basisOfProcessing": {
+          "$ref": "#/definitions/basisOfProcessing"
+        },
+        "xdm:timestamp": {
+          "$ref": "#/definitions/timestamp"
+        },
+        "xdm:source": {
+          "$ref": "#/definitions/source"
+        },
+        "xdm:reason": {
+          "title": "Opt out reason",
+          "type": "string",
+          "description": "Reason for opting out",
+          "maxLength": 20
+        }
+      }
+    },
+    "consent-options": {
+      "properties": {
+        "xdm:privacyPreferences": {
+          "title": "Privacy Preferences",
+          "description": "Encapsulates data privacy preferences or consents.",
+          "type": "object",
+          "properties": {
+            "xdm:dataCollection": {
+              "title": "Data Collection",
+              "description": "Data collection is permitted",
+              "$ref": "#/definitions/consentOption"
+            },
+            "xdm:sellData": {
+              "title": "Sell Data",
+              "description": "Selling of user's Data with/to 2nd or 3rd parties is permitted, such as in compliance with CCPA",
+              "$ref": "#/definitions/consentOption"
+            },
+            "xdm:shareData": {
```

Comment:
@RussStringham  - Does this need to account for second party vs third party sharing?

Reply:
end users don't understand the difference between the two types of sharing, so it would be difficult to ask a question that could distinguish between the two. Also, CCPA doesn't distinguish.

Reply:
I distinctly remember recently responding to a privacy notice where it asked if it was ok for my data to be shared with affiliates within the company (2P) and outside the company (3P). They were different choices. Could you please check with legal/PM on this. While the definition of 2P/3P may be less relevant here - the important question whether users can tell Unilever that they cannot share data with Amex vs cannot share data within Unilever brand companies.

Reply:
That's not the definition of 2nd versus 3rd party. 2nd party is where the company shares the data directly with another entity (affiliate or other company). 3rd party is where you share with an intermediary and they share with other companies, or more generally where the 2nd party shares the data shared with them with others.

Reply:
You are correct that banking regulations restrict some 1st-party sharing (including affiliates) without permission, such as details about your checking account being shared with the credit-card division for sending offers for new credit cards based on how much money moves through your checking account.
I'll talk to the PM about adding this use case.

Reply:
Sure. 2P vs 3P distinction is not my main concern. The important question is whether users can tell Unilever that they cannot share data with Amex vs cannot share data within Unilever brand companies. I have seen multiple privacy data sharing forms that make a distinction between these 2 kinds of sharing and I am wondering if that applies to AEP consent collection.

Reply:
My thoughts: If the different Unilever brands use different IMS Orgs for collecting data, then the xdm:dataSharing consent would apply, just as it does for 2nd/3rd-parties. With demand, we could add a new consent to distinguish between these types of sharing.
If they are part of the same IMS Org, then we would need significant changes to all products to keep the data separate for user's who make this choice. For example, for every query we would need to know the organization making the query, the organization responsible for the experience event or profile attribute and whether sharing was allowed between the two.

Reply:
Ok. We can add more cases when the need arise to differentiate between different kinds of sharing specifically. The current value can be understood as allow all kinds of sharing.